src/lib/storage/index/adaptive_radix_tree/adaptive_radix_tree_index.hpp

```diff
@@ -37,6 +37,12 @@ class AdaptiveRadixTreeIndex : public BaseIndex {
   friend class AdaptiveRadixTreeIndexTest_BinaryComparableFromChunkOffset_Test;
 
  public:
+  /**
+   * Predicts the memory consumption in bytes of creating this index.
+   * See BaseIndex::predict_memory_consumption()
+   */
+  static uintptr_t predict_memory_consumption(ChunkOffset row_count, ChunkOffset value_count, uint32_t value_bytes);
```

Comment:
Why `uintptr_t`? This is a size. `size_t` has it right in the name. ;)

Reply:
Our motivation was that `uintptr_t` can actually be larger than `size_t` on some platforms so it's possible to use more memory than can be represented by a `size_t`. But I guess that in these cases the index will probably be limited to the size of `size_t` in reasonable implementations. So changed to `size_t`.